include/zephyr/net/wifi_mgmt.h

```diff
@@ -41,6 +41,12 @@ extern "C" {
 #define WIFI_MGMT_SCAN_SSID_FILT_MAX 0
 #endif /* CONFIG_WIFI_MGMT_SCAN_SSID_FILT_MAX */
 
+#ifdef CONFIG_WIFI_MGMT_SCAN_CHAN_MAX_MANUAL
+#define WIFI_MGMT_SCAN_CHAN_MAX_MANUAL CONFIG_WIFI_MGMT_SCAN_CHAN_MAX_MANUAL
+#else
+#define WIFI_MGMT_SCAN_CHAN_MAX_MANUAL 1
+#endif /* CONFIG_WIFI_MGMT_SCAN_CHAN_MAX_MANUAL */
```

Comment:
In which case we hit the else branch? Can there be some other `wifi_mgmt` implementation than the one in `subsys/net/l2/wifi`?

Reply:
This is a workaround for a bug in few cellular/location samples, the Wi-Fi headers are included unconditionally (`nrf/include/net/wifi_location_common.h`) and if the Wi-Fi isn't enabled, this poses a problem, as Kconfig isn't included. This was first instance of a Kconfig being used in the header files.

Reply:
Thanks for the info. I know some/many people prefer to use as little conditions as possible which causes these issues. I'll keep this in mind to see if there would be better solution for this.